Orlando Walking Tours/Controllers/LocationListVC.swift

```diff
@@ -7,26 +7,284 @@
+            let locationId = cellView.locationId
+            let locationIndex = self.locations.indexOf { $0.locationId == locationId }!
+            print("add to tour \(locationIndex)")
+            let location = self.locations.removeAtIndex(locationIndex)
+            modelService.addLocation(location, toTour: self.tour!)
+            { (ok, error) in
+                if ok
+                {
+                    let indexPath = NSIndexPath(forItem: locationIndex, inSection: 0)
+                    self.tableView.deleteRowsAtIndexPaths([indexPath], withRowAnimation: .Right)
+                    self.displayTour()
+                }
+                else
+                {
+                    // !!! need to handle errors here
+                }
+            }
+        }
+    }
+
+    ////////////////////////////////////////////////////////////
+    
+    @IBAction func listViewTapped(sender: UIButton)
+    {
+        self.viewMode = .List
+    }
+
+    ////////////////////////////////////////////////////////////
+
+    @IBAction func mapViewTapped(sender: UIButton)
+    {
+        self.viewMode = .Map
+    }
+
+    ////////////////////////////////////////////////////////////
+    // MARK: - Helper Functions
+    ////////////////////////////////////////////////////////////
+
+    func setAvailableTourLocations(allLocations: [HistoricLocation])
+    {
+        if let tourLocations = self.tour?.historicLocations
+        {
+            for loc in tourLocations
+            {
+                print("LOC: \(loc.locationTitle)")
+            }
+            var availableTourLocations = [HistoricLocation]()
+            for tourLoc in allLocations
+            {
+                if !tourLocations.contains({ $0.locationTitle == tourLoc.locationTitle
+                }) {
+                    availableTourLocations.append(tourLoc)
+                }
+            }
+            self.locations = availableTourLocations
+        }
+        else
+        {
+            self.locations = allLocations
+        }
+    }
+
+    ////////////////////////////////////////////////////////////
+    
+    func sortClosestToMe()
+    {
+        if let userLocation = self.userLocation
+        {
+            // first collect loc and distance together in a tuple
+            let locDist = locations.map
+            { loc -> (CLLocationDistance, HistoricLocation) in
+                let siteLoc = loc.locationPoint
+                let distanceFromCur = siteLoc.distanceFromLocation(userLocation).asMiles
+                return (distanceFromCur, loc)
+            }
+            // now sort by distance
+            self.locations = locDist.sort
+            { (a,b) in
+                return a.0 <= b.0
+            }
+            // and finally pull out the location
+            .map
+            { (dist, loc) in
+                return loc
+            }
+        }
+    }
+
+    ////////////////////////////////////////////////////////////
+
+    func displayTour()
```

Comment:
@gregsbarr is this basically just a debugging function?  to show the list of locations that have been added to the tour so far?